File: src/Entity/Stagiaire.php
<?php

namespace App\Entity;

use App\Repository\StagiaireRepository;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=StagiaireRepository::class)
 */
class Stagiaire
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $Etablissement;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $TypeDiplome;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $filiereDiplome;

    /**
     * @ORM\Column(type="date")
     */
    private $DateDiplome;

    /**
     * @ORM\Column(type="integer")
     */
    private $Experience;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $Competence;

    /**
     * @ORM\OneToOne(targetEntity=CandidatStage::class, mappedBy="IDStagiaire", cascade={"persist", "remove"})
     */
    private $candidatStage;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getEtablissement(): ?string
    {
        return $this->Etablissement;
    }

    public function setEtablissement(string $Etablissement): self
    {
        $this->Etablissement = $Etablissement;

        return $this;
    }

    public function getTypeDiplome(): ?string
    {
        return $this->TypeDiplome;
    }

    public function setTypeDiplome(string $TypeDiplome): self
    {
        $this->TypeDiplome = $TypeDiplome;

        return $this;
    }

    public function getFiliereDiplome(): ?string
    {
        return $this->filiereDiplome;
    }

    public function setFiliereDiplome(string $filiereDiplome): self
    {
        $this->filiereDiplome = $filiereDiplome;

        return $this;
    }

    public function getDateDiplome(): ?\DateTimeInterface
    {
        return $this->DateDiplome;
    }

    public function setDateDiplome(\DateTimeInterface $DateDiplome): self
    {
        $this->DateDiplome = $DateDiplome;

        return $this;
    }

    public function getExperience(): ?int
    {
        return $this->Experience;
    }

    public function setExperience(int $Experience): self
    {
        $this->Experience = $Experience;

        return $this;
    }

    public function getCompetence(): ?string
    {
        return $this->Competence;
    }

    public function setCompetence(string $Competence): self
    {
        $this->Competence = $Competence;

        return $this;
    }

    public function getCandidatStage(): ?CandidatStage
    {
        return $this->candidatStage;
    }

    public function setCandidatStage(CandidatStage $candidatStage): self
    {
        // set the owning side of the relation if necessary
        if ($candidatStage->getIDStagiaire() !== $this) {
            $candidatStage->setIDStagiaire($this);
        }

        $this->candidatStage = $candidatStage;

        return $this;
    }
}

File: src/models/Stagiaire.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package models;

import java.sql.Date;

/**
 *
 * @author User
 */
public class Stagiaire {
     private Integer id;

    private String etablissemnt;
 
    
    private String type_diplome;

  
    private String filiere_diplome;

  
    private Date date_diplome;

   
    private Integer experience;

  
    private String competence;
    
    private String nom;

    public Stagiaire() {
    }

    public Stagiaire(String etablissemnt, String type_diplome, String filiere_diplome, Date date_diplome, Integer experience, String competence, String nom) {
        this.etablissemnt = etablissemnt;
        this.type_diplome = type_diplome;
        this.filiere_diplome = filiere_diplome;
        this.date_diplome = date_diplome;
        this.experience = experience;
        this.competence = competence;
        this.nom = nom;
    }

    public Stagiaire(Integer id, String etablissemnt, String type_diplome, String filiere_diplome, Date date_diplome, Integer experience, String competence, String nom) {
        this.id = id;
        this.etablissemnt = etablissemnt;
        this.type_diplome = type_diplome;
        this.filiere_diplome = filiere_diplome;
        this.date_diplome = date_diplome;
        this.experience = experience;
        this.competence = competence;
        this.nom = nom;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getEtablissemnt() {
        return etablissemnt;
    }

    public void setEtablissemnt(String etablissemnt) {
        this.etablissemnt = etablissemnt;
    }

    public String getType_diplome() {
        return type_diplome;
    }

    public void setType_diplome(String type_diplome) {
        this.type_diplome = type_diplome;
    }

    public String getFiliere_diplome() {
        return filiere_diplome;
    }

    public void setFiliere_diplome(String filiere_diplome) {
        this.filiere_diplome = filiere_diplome;
    }

    public Date getDate_diplome() {
        return date_diplome;
    }

    public void setDate_diplome(Date date_diplome) {
        this.date_diplome = date_diplome;
    }

    public Integer getExperience() {
        return experience;
    }

    public void setExperience(Integer experience) {
        this.experience = experience;
    }

    public String getCompetence() {
        return competence;
    }

    public void setCompetence(String competence) {
        this.competence = competence;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    @Override
    public String toString() {
        return + id + ":" + nom;
    }
    
    
}
